File: Examples/FragmentDynamicLayout/src/course/examples/Fragments/DynamicLayout/TitlesFragment.java
package course.examples.Fragments.DynamicLayout;

import android.app.Activity;
import android.app.ListFragment;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class TitlesFragment extends ListFragment {
	private static final String TAG = "TitlesFragment";
	private ListSelectionListener mListener = null;

	public interface ListSelectionListener {
		public void onListSelection(int index);
	}

	@Override
	public void onListItemClick(ListView l, View v, int pos, long id) {
		getListView().setItemChecked(pos, true);
		mListener.onListSelection(pos);
	}

	@Override
	public void onAttach(Activity activity) {
		Log.i(TAG, getClass().getSimpleName() + ":entered onAttach()");
		super.onAttach(activity);
		
		try {
			mListener = (ListSelectionListener) activity;
		} catch (ClassCastException e) {
			throw new ClassCastException(activity.toString()
					+ " must implement OnArticleSelectedListener");
		}
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		Log.i(TAG, getClass().getSimpleName() + ":entered onCreate()");
		super.onCreate(savedInstanceState);
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
		Log.i(TAG, getClass().getSimpleName() + ":entered onCreate()");
		return super.onCreateView(inflater, container, savedInstanceState);
	}


	@Override
	public void onActivityCreated(Bundle savedState) {
		Log.i(TAG, getClass().getSimpleName() + ":entered onActivityCreated()");
		super.onActivityCreated(savedState);
		
		setListAdapter(new ArrayAdapter<String>(getActivity(),
				R.layout.title_item, QuoteViewerActivity.mTitleArray));
		getListView().setChoiceMode(ListView.CHOICE_MODE_SINGLE);
	}

	@Override
	public void onStart() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onStart()");
		super.onStart();
	}

	@Override
	public void onResume() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onResume()");
		super.onResume();
	}

	@Override
	public void onPause() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onPause()");
		super.onPause();
	}

	@Override
	public void onStop() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onStop()");
		super.onStop();
	}

	@Override
	public void onDetach() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onDetach()");
		super.onDetach();
	}

	@Override
	public void onDestroy() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onDestroy()");
		super.onDestroy();
	}

	@Override
	public void onDestroyView() {
		Log.i(TAG, getClass().getSimpleName() + ":entered onDestroyView()");
		super.onDestroyView();
	}

}

File: ExamplesKotlin/FragmentDynamicLayout/app/src/main/java/course/examples/fragments/dynamiclayout/TitlesFragment.kt
package course.examples.fragments.dynamiclayout

import android.content.Context
import android.os.Bundle
import android.support.v4.app.ListFragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView

//Several Activity and Fragment lifecycle methods are instrumented to emit LogCat output
//so you can follow the class' lifecycle
class TitlesFragment : ListFragment() {

    companion object {
        private const val TAG = "TitlesFragment"
    }

    private var mListener: ListSelectionListener? = null
    private var mCurrIdx = ListView.INVALID_POSITION


    fun unCheckSelection() {
        if (listView.checkedItemCount > 0) {
            listView.setItemChecked(listView.checkedItemPosition, false)
        }
        mCurrIdx = ListView.INVALID_POSITION
    }

    // Called when the user selects an item from the List
    override fun onListItemClick(l: ListView, v: View, pos: Int, id: Long) {
        if (mCurrIdx != pos) {
            mCurrIdx = pos
            // Indicates the selected item has been checked
            l.setItemChecked(mCurrIdx, true)
            // Inform the QuoteViewerActivity that the item in position pos has been selected
            mListener?.onListSelection(mCurrIdx)
        }
    }


    override fun onAttach(context: Context) {
        Log.i(TAG, "${javaClass.simpleName}: entered onAttach()")
        super.onAttach(context)

        // Set the ListSelectionListener for communicating with the QuoteViewerActivity
        mListener = context as? ListSelectionListener

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Log.i(TAG, "${javaClass.simpleName}: entered onCreate()")
        super.onCreate(savedInstanceState)

        retainInstance = true
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        Log.i(TAG, "${javaClass.simpleName}: entered onCreateView()")
        return super.onCreateView(inflater, container, savedInstanceState)
    }


    override fun onActivityCreated(savedState: Bundle?) {
        Log.i(TAG, "${javaClass.simpleName}: entered onActivityCreated()")
        super.onActivityCreated(savedState)

        // Set the list adapter for the ListView
        // Discussed in more detail in the user interface classes lesson
        listAdapter = ArrayAdapter(
            activity as Context,
            R.layout.title_item, QuoteViewerActivity.mTitleArray
        )

        // Set the list choice mode to allow only one selection at a time
        listView.choiceMode = ListView.CHOICE_MODE_SINGLE
    }

    override fun onStart() {
        Log.i(TAG, "${javaClass.simpleName}: entered onStart()")
        super.onStart()
    }

    override fun onResume() {
        Log.i(TAG, "${javaClass.simpleName}: entered onResume()")
        super.onResume()
    }

    override fun onPause() {
        Log.i(TAG, "${javaClass.simpleName}: entered onPause()")
        super.onPause()
    }

    override fun onStop() {
        Log.i(TAG, "${javaClass.simpleName}: entered onStop()")
        super.onStop()
    }

    override fun onDetach() {
        Log.i(TAG, "${javaClass.simpleName}: entered onDetach()")
        super.onDetach()
    }

    override fun onDestroy() {
        Log.i(TAG, "${javaClass.simpleName}: entered onDestroy()")
        super.onDestroy()
    }

    override fun onDestroyView() {
        Log.i(TAG, "${javaClass.simpleName}: entered onDestroyView()")
        super.onDestroyView()
    }
}